### PR TITLE
feat: add transfer and approval logs

### PR DESCRIPTION
**Pull request by 0xShishigami**
## Describe your changes
- Load Transfer and Approval logs by selected Token
- Display logs in a data grid

## Issue ticket number and link
[Implement account history of transfer and approvals](https://linear.app/defi-wonderland/issue/FEO-10/implement-account-history-of-transfers-and-approvals)
closes FEO-10

## Checklist before requesting a review

- [x] I have conducted a self-review of my code.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,6 +38,7 @@
     "@mui/icons-material": "5.15.6",
     "@mui/material": "5.15.14",
     "@mui/material-nextjs": "5.15.11",
+    "@mui/x-data-grid": "^7.15.0",
```
Comment (0xArdy) [inline package.json]: for security we recommend to lock de dependency versions

**Review by 0xArdy — APPROVED** [review]
what a clean implementation! nice work! 🔥 

If I can comment on something, I would recommend defining the blockchain's read and write functions outside of the components, in a utils folder for example, but up to u 😄

File: src/providers/TokenProvider.tsx [hunk 2]
```diff
@@ -184,10 +208,66 @@ export const TokenProvider = ({ children }: TokenProps) => {
...
+      address: tokenSelected.address,
+      event: WONDER_TOKEN_EVENT_APPROVAL_ABI,
+      args: { owner: address },
+      fromBlock: 'earliest',
+      toBlock: 'latest',
+    });
+
+    const tmpLogs = [...transferLogs, ...approvalLogs];
+
+    const sortedLogs = tmpLogs.sort((a, b) => {
```
Comment (0xtiti) [inline src/providers/TokenProvider.tsx:232]: I think you could try using a generic type `const sortedLogs = tmpLogs.sort<EventLogs>((a, b) => {` and `setLogs(sortedLogs);`

**Review by 0xtiti — APPROVED** [review]

File: src/providers/TokenProvider.tsx (hunk 2 above)
Comment (0xShishigami) [reply src/providers/TokenProvider.tsx:232]: Doesn't seem to work. 
The typing here is just to match the expected object in the UI (ex. log.args.spender), that's why it ended with an unknown and a custom type.
I tried using the native type Log<typeof ABI_EVENT> and the GetLogsReturnType, getting nothing but TS mismatch hell.

Screenshot 2024-09-05 at 9 15 12 PM

File: package.json (hunk 1 above)
Comment (0xShishigami) [reply package.json]: https://github.com/0xShishigami/feo-web3-react-boilerplate/pull/5/commits/169648a4487575eeb84eb6cb972c48cff4fa646d

**Pull request merged**